src/foolscap/test/test_banana.py

```diff
@@ -398,60 +478,63 @@ def test_aborted_list3(self):
                                tOPEN(2),'list', 5, 6, tABORT, tCLOSE(2),
                               tCLOSE(1),
                              tCLOSE(0)])
-        self.failUnless(isinstance(f, BananaFailure))
-        self.failUnless(f.check(Violation))
-        self.failUnlessEqual(f.value.where, "<RootUnslicer>.[1].[3].[2]")
-        self.failUnlessEqual(f.value.args[0], "ABORT received")
+        self.assertTrue(isinstance(f, BananaFailure))
+        self.assertTrue(f.check(Violation))
+        self.assertEqual(f.value.where, "<RootUnslicer>.[1].[3].[2]")
+        self.assertEqual(f.value.args[0], "ABORT received")
 
     def test_nested_list(self):
         "nested list"
         res = self.do([tOPEN(0),'list',1,2,
                         tOPEN(1),'list',3,4,tCLOSE(1),
                        tCLOSE(0)])
-        self.failUnlessEqual(res, [1,2,[3,4]])
+        self.assertEqual(res, [1,2,[3,4]])
 
     def test_list_with_tuple(self):
         "list with tuple"
         res = self.do([tOPEN(0),'list',1,2,
                         tOPEN(1),'tuple',3,4,tCLOSE(1),
                        tCLOSE(0)])
-        self.failUnlessEqual(res, [1,2,(3,4)])
+        self.assertEqual(res, [1,2,(3,4)])
 
     def test_dict(self):
         "dict"
         res = self.do([tOPEN(0),'dict',"a",1,"b",2,tCLOSE(0)])
-        self.failUnlessEqual(res, {'a':1, 'b':2})
+        self.assertEqual(res, dict_key_to_bytes({'a':1, 'b':2}))
 
     def test_dict_with_duplicate_keys(self):
         "dict with duplicate keys"
         f = self.shouldDropConnection([tOPEN(0),'dict',
                                        "a",1,"a",2,
                                        tCLOSE(0)])
-        self.failUnlessEqual(f.value.where, "<RootUnslicer>.{}")
-        self.failUnlessEqual(f.value.args[0], "duplicate key 'a'")
+        self.assertEqual(f.value.where, "<RootUnslicer>.{}")
+        if six.PY2:
+            self.assertEqual(f.value.args[0], "duplicate key 'a'")
+        elif six.PY3:
+            self.assertEqual(f.value.args[0], "duplicate key 'b'a''")       
 
```

Comment:
Instead it's probably better to ensure we give consistent error-message across py2/3